### ISSUE TITLE
Different buttons for input with modifier keys

### ISSUE THREAD

**Issue by jozefmutis** [univrsal/input-overlay#73]
**What it is**
A two separate buttons would exist, and each light up individually depending on what modifier is used.

**Example**
Should there be a button for Left Mouse Click, and a button for Shift + Left Mouse Click. One button would light up only if LMC was pressed by itself, and another button would light up if LMC was pressed in combination with SHIFT modifier

**How it can work**
A normal Left Mouse Button icon would light up only if it detects that SHIFT key is not pressed. This could be achieved by adding blacklist keys as a config in options.ini. A button will not light up if it detects any of the inputs defined as blacklisted are currently pressed.

A Shift LMC would light up only if it detects that SHIFT key AND Left Mouse Click are pressed concurrently. This can be achieved by checking for multiple key inputs before allowing button to light up.

**Additional context**
This would of course not impact the button for SHIFT key itself, as it would light up regardless whether there was a follow-up LMC click registered.

Of course this could be applied to inputs other than Left Mouse Click, as configurable in options.ini.

**Comment by univrsal** [comment]
So basically you want an overlay element that handles input combinations? Like an element that only changed from unpressed to pressed if a key combination is pressed (e.g. Ctr+V or in your example Shift + Left Mouse)

I already got this suggestion a while back but didn't consider it since it would've been more complicated, but I think since I made a lot of changes it shouldn't be too difficult anymore.

**Comment by jozefmutis** [comment]
Yes, exactly. Along with this, I believe you could also add function so that some buttons work toggle only if no key modifiers are used (e.g. plain V, and would not light up if Ctr+V was pressed)

I feel this could easily be achieved via blacklist and checking for multiple inputs, but if you have a solution of your own, please don't be limited by mi imagination.

**Comment by GGelatin** [comment]
Was wondering if there's an update on this issue?
I tried to create my own preset but was unable to figure out a way to create combination keycodes.

**Comment by univrsal** [comment]
No, there's still some other things that need to be addressed first

**Comment by drdecomposing** [comment]
Came looking for something like this — any updates on this issue? This kind of feature would be enormously helpful to folks who use this plugin with alternative/non-standard controllers.
